bson/src/main/org/bson/codecs/configuration/mapper/ClassModel.java

```diff
@@ -34,6 +35,10 @@
     private final Map<String, FieldModel> fields = new TreeMap<String, FieldModel>();
     private final CodecRegistry registry;
     private final TypeResolver resolver;
+    private final MemberResolver memberResolver;
+    private final WeightedValue<String> collectionName = new WeightedValue<String>();
```

Comment:
That's a great point actually.  With the driver, we'd have an explicit MongoCollection in hand.  This name is really for ODMs (read:  Morphia) to know how to get the appropriate MongoCollection to use.  hrm...